framework/test/jest/unit/specs/modules/chain/synchronizer/block_synchronization_mechanism.spec.js

```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright © 2018 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+'use strict';
+
+const BlockSynchronizationMechanism = require('../../../../../../../src/modules/chain/synchronizer/block_synchronization_mechanism');
+
+describe('BlockSynchronizationMechanism', () => {
+	const stubs = {
+		channel: {
+			invoke: jest.fn(),
+		},
+		modules: {
+			blocks: {
+				lastBlock: jest.fn(),
+			},
+		},
+	};
+	let blockSynchronizationMechanism;
+
+	beforeEach(() => {
+		blockSynchronizationMechanism = new BlockSynchronizationMechanism({
+			channel: stubs.channel,
+			modules: stubs.modules,
+		});
+	});
+
+	describe('#constructor', () => {});
```

Comment:
why constructor test suite is empty?

Reply:
Added the skeleton as these tests will be completed once we have both mechanism in place.